test/distributed/_shard/sharded_tensor/test_sharded_tensor_matrix_ops.py

```diff
@@ -0,0 +1,108 @@
+# Owner(s): ["oncall: distributed"]
+
+import sys
+
+import torch
+from torch.distributed._shard import (
+    sharded_tensor,
+)
+from torch.testing._internal.common_distributed import (
+    requires_nccl,
+    skip_if_lt_x_gpu,
+)
+from torch.testing._internal.common_utils import (
+    TEST_WITH_DEV_DBG_ASAN,
+    run_tests,
+)
+from torch.testing._internal.distributed._shard.sharded_tensor import (
+    ShardedTensorTestBase,
+    with_comms,
+)
+from torch.testing._internal.distributed._shard.sharded_tensor._test_st_common import (
+    _chunk_sharding_specs_list_for_test,
+)
+
+if TEST_WITH_DEV_DBG_ASAN:
+    print(
+        "Skip dev-asan as torch + multiprocessing spawn have known issues",
+        file=sys.stderr,
+    )
+    sys.exit(0)
+
+
+class TestShardedTernsorMatrixOps(ShardedTensorTestBase):
+    @with_comms(init_rpc=True)
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_sharded_tensor_transpose(self):
+        specs = _chunk_sharding_specs_list_for_test([0], seed=7)
+        for spec in specs:
+            st = sharded_tensor.rand(spec, 10, 20, 5, init_rrefs=True)
+            self.assertEqual(3, st.dim())
+            self.assertEqual(torch.Size([10, 20, 5]), st.size())
+            local_tensor = st.local_tensor()
+            st = st.transpose(1, 0)
+            spec.dim = 1
+            local_tensor = local_tensor.transpose(1, 0)
+            st_2 = sharded_tensor.rand(spec, 20, 10, 5, init_rrefs=True)
+            self.assertEqual(st_2.size(), st.size())
+            self.assertEqual(local_tensor, st.local_tensor())
+            self.assertEqual(st_2.metadata(), st.metadata())
```

Comment:
We are not really testing local and sharded version of the ops are same here since we are pulling out st.local_tensor(). We should run the op with a local tensor, shard the same local tensor and run the op and verify results.

Reply:
sure. Let me add that test cases.